### DISCUSSION TITLE
Prepare for v4.2

### DISCUSSION THREAD

**Discussion by axunonb** [YAXLib/YAXLib#231] [General]
@ycherkes:
Now we have the `version/4.2. branch.

Making a built-in type like `DateTime` a known type isn't that simple, unfortunately. The current implementation let's you use many nice attributes like `YAXSerializeAsAttribute` or `YAXAttributeForClassAttribute` for them.
For an `IKnownType` we just have the `Serialize` and `Deserialize` methods. The heavy lifting for attributes would have to be done there. Any ideas?

Feel free to add PRs for non-breaking features to the `version/4.2` branch. Make sure to target this branch, please.

**Comment by axunonb** [comment]
@ycherkes: Is there anything from your side which could go into v4.2? 
Otherwise we could release v4.2.
Also, it would be great if you joined YAXLib as a co-maintainer if your time allows.

**Reply by ycherkes** [reply]
@axunonb: Thank you for the invitation.
Unfortunately, I'm not sure if I'll be able to work on the project regularly in the near future.
I think we could release 4.2 because changing the DateTime serialization is a very difficult task.

**Reply by axunonb** [reply]
Okay, let's go for it. Great job BTW! Would you like to draft the release notes?

**Reply by ycherkes** [reply]
@axunonb Please take a look at the fix: [PR!236](https://github.com/YAXLib/YAXLib/pull/236)

**Comment by ycherkes** [comment]
Release notes:

v4.2.0
What's Changed

1. Add the **DoNotSerializeDefaultValues** option to **YAXSerializationOptions**.

2. Add the **TypeInspector** property to **SerializerOptions**.

3. Mark **PropertyInfo**, **FieldInfo**, and **MemberInfo** properties as obsolete in **IMemberContext**.

4. Add the **MemberDescriptor** property to **IMemberContext** as a replacement for deprecated properties.

5. Miscellaneous:

* Update MemberWrapperCache and UdtWrapperCache: cache the type by type and SerializerOptions.
* Remove usage of Lazy<T> & ThreadLocal<T> as they complicate using serializer in Debug mode [ "The function evaluation requires all threads to run"](https://stackoverflow.com/questions/4460206/lazyt-the-function-evaluation-requires-all-threads-to-run).

**Reply by axunonb** [reply]
@ycherkes: Great job, thanks a lot! v4.2.0 is now published.
Even if you'll not be able to work on the project regularly, I'd still appreciate if you contributed by reviewing PRs.
